core/src/broadcast_stage/standard_broadcast_run.rs

```diff
@@ -344,7 +344,7 @@ impl StandardBroadcastRun {
         if now - last > BROADCAST_PEER_UPDATE_INTERVAL_MS
             && self
                 .last_peer_update
-                .compare_and_swap(now, last, Ordering::Relaxed)
```

Comment:
https://doc.rust-lang.org/std/sync/atomic/struct.AtomicUsize.html#method.compare_and_swap:

```
pub fn compare_and_swap(
    &self,
    current: usize,
    new: usize,
    order: Ordering
) -> usize

Use compare_exchange or compare_exchange_weak instead

Stores a value into the atomic integer if the current value is the same as the current value.
```

Notice we're passing `now` as `current`, which means cas never succeeds and `.compare_and_swap()` is returning `last` value always, then `if` always evaluating to `true` here (kicking the ClusterNodes calc.